integrationtests/test_all_examples.py

```diff
@@ -0,0 +1,281 @@
+
+    Parameters
+    ----------
+    directory : str
+        Path of the directory to be walked.
+
+    Returns
+    -------
+    paths : list of str
+        List of Python file paths.
+    """
+    paths = []
+    for root, _, files in os.walk(directory):
+        for filename in files:
+            path = os.path.abspath(os.path.join(root, filename))
+            if is_python_file(path):
+                paths.append(path)
+    return sorted(paths)
+
+
+def replaced_configure_traits(
+        instance,
+        filename=None,
+        view=None,
+        kind=None,
+        edit=True,
+        context=None,
+        handler=None,
+        id="",
+        scrollable=None,
+        **args
+):
+    """ Mocked configure_traits to launch then close the GUI.
+    """
+    ui = instance.edit_traits(
+        view=view,
+        parent=None,
+        kind="live",  # other options may block the test
+        context=context,
+        handler=handler,
+        id=id,
+        scrollable=scrollable,
+        **args,
+    )
+    with store_exceptions_on_all_threads():
+        GUI.process_events()
+
+        # Temporary fix for enthought/traitsui#907
+        if is_current_backend_qt4():
+            ui.control.hide()
+        if is_current_backend_wx():
+            ui.control.Hide()
+
+        ui.dispose()
+        GUI.process_events()
+
+
+@contextlib.contextmanager
+def replace_configure_traits():
+    """ Context manager to temporarily replace HasTraits.configure_traits
+    with a mocked version such that GUI launched are closed soon after they
+    are open.
+    """
+    original_func = HasTraits.configure_traits
+    HasTraits.configure_traits = replaced_configure_traits
+    try:
+        yield
+    finally:
+        HasTraits.configure_traits = original_func
+
+
+def run_file(file_path):
+    """ Execute a given Python file.
+    Parameters
+    ----------
+    file_path : str
+        File path to be tested.
+    """
+    with open(file_path, "r", encoding="utf-8") as f:
+        content = f.read()
+
+    globals = {
+        "__name__": "__main__",
+        "__file__": file_path,
+    }
+    with replace_configure_traits(), \
+            mock.patch("sys.stdout", new_callable=io.StringIO), \
+            mock.patch("sys.argv", [file_path]):
+        # Mock stdout: Examples typically print educational information.
+        # They are expected but they should not pollute test output.
+        # Move argv: Some example reads sys.argv to allow more arguments
+        # But all examples should support being run without additional
+        # arguments.
+        exec(content, globals)
+
+
+@skip_if_null
+class TestExample(unittest.TestCase):
+
+    def test_run(self):
+        check_special_files()
+        file_paths = chain.from_iterable(
+            get_python_files(source_dir) for source_dir in SOURCE_DIRS
+        )
+        for file_path in file_paths:
+            with self.subTest(file_path=file_path):
```

Comment:
I don't think we want logging here. We typically want to use the test runner flag to determine verbosity. In most situations, tests should not unconditionally emit logs that will pollute the console (TraitsUI test suite is not in a clean state in that regard).

Reply:
This might be relevant: https://bugs.python.org/issue35327

Reply:
The way I see it is that each example should have its own test and therefore should be logged just like other tests (not subtests). With that it would be very clear how many examples tests have run, in what order they were run, how many were skipped or have failed (if it's a big proportion of existing examples), which examples are being noisy, which ones have uncaught exceptions, etc... 

I agree that verbosity should be controlled via the test runner flag and after quick googling it seems that the flag can't be accessed from within a `TestCase`.

Given that, maybe `subTest` isn't the best approach here, maybe generating tests via a metaclass would work better?

Reply:
Ah... I think that would be an overkill to workaround some missing feature of unittest test runner. Let's get the examples covered by tests and then we can make the test machinery better.